torch/testing/_internal/common_methods_invocations.py

```diff
@@ -17531,6 +17531,27 @@ def __init__(
         ukwargs = _inherit_constructor_args(name, op, inherited, kwargs)
         super(OpInfoPythonRefInfo, self).__init__(**ukwargs)
 
+class ReductionPythonRefInfo(ReductionOpInfo):
+    '''
+    An OpInfo for a Python reference of an elementwise unary operation.
+    '''
+    def __init__(
+            self,
+            name,  # the stringname of the callable Python reference
+            *,
+            op=None,  # the function variant of the operation, populated as torch.<name> if None
+            torch_opinfo_name,  # the string name of the corresponding torch opinfo
+            **kwargs):  # additional kwargs override kwargs inherited from the torch opinfo
+
+        self.torch_opinfo_name = torch_opinfo_name
+        self.torch_opinfo = _find_referenced_opinfo(torch_opinfo_name)
+        assert isinstance(self.torch_opinfo, ReductionOpInfo)
+
+        inherited = self.torch_opinfo._original_opinfo_args
+        ukwargs = _inherit_constructor_args(name, op, inherited, kwargs)
+
+        super(ReductionPythonRefInfo, self).__init__(**ukwargs)
```

Comment:
No need to do old-style Python2 super call, just super().__init__

Reply:
Also, now that the base OpInfo class is a dataclass, the subclasses of OpInfo ideally should also be declared as dataclasses too

cc @francescocastelli